llvm/lib/CodeGen/LiveDebugValues/VarLocBasedImpl.cpp

```diff
@@ -1825,10 +1825,9 @@ static bool isSwiftAsyncContext(const MachineInstr &MI) {
   unsigned Reg = isDbgValueDescribedByReg(MI);
   if (!Reg)
     return false;
-  auto &EntryMBB = MF->front();
   unsigned I = 0;
-  for (auto R : EntryMBB.liveins()) {
-    if (R.PhysReg == Reg && F.hasParamAttribute(I, Attribute::SwiftAsync))
+  for (auto R : MF->getRegInfo().liveins()) {
+    if (R.first == Reg && F.hasParamAttribute(I, Attribute::SwiftAsync))
```

Comment:
Might be clearer to do the `F.hasParamAttribute(I, Attribute::SwiftAsync)` check above, where we check `F.getCallingConv()`.

Reply:
Just curious - the only `liveins` field I see in the MIR is in the entry MBB; where do we pick up the extra liveins?

Reply:
Here is the concrete (and surprising!) example MIR that prompted this:

```
Function Live Ins: $x22, $x0
bb.0 (%ir-block.2):
  successors: %bb.2, %bb.1
  liveins: $x0, $x22, $lr
```

Reply:
Weird, so we actually want the extra livein for $lr?
Is it legitimate to exclude $lr from the MF's liveins list?
Also, what's up with the ordering of the liveins? Is it random? If $lr were the first entry in the liveins list, couldn't the `F.hasParamAttribute(I, Attribute::SwiftAsync)` check spuriously fail?

Reply:
@aemerson @ornata @TNorthover  Do you what the best way would be to query which incoming register corresponds to which llvm::Argument? The condition I want to write here is: "is this the incoming register that the ABI reserves for the swift async context?"